crates/aptos/src/move_tool/disassembler.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use move_command_line_common::files::{
+    MOVE_COMPILED_EXTENSION, MOVE_EXTENSION, SOURCE_MAP_EXTENSION,
+};
+use crate::common::types::{CliCommand, CliTypedResult};
+use std::{fs, path::Path};
+use move_binary_format::binary_views::BinaryIndexedView;
+use move_binary_format::CompiledModule;
+use move_binary_format::file_format::CompiledScript;
+use move_coverage::coverage_map::CoverageMap;
+use move_disassembler::disassembler::{Disassembler, DisassemblerOptions};
+use clap::Parser;
+use async_trait::async_trait;
+use move_bytecode_source_map::{mapping::SourceMapping, utils::source_map_from_file};
+use move_ir_types::location::Spanned;
+
+/// Disassemble the Move bytecode pointed to
+#[derive(Debug, Parser)]
+pub struct AptosDisassembler {
+    /// Skip printing of private functions.
+    #[clap(long = "skip-private")]
+    pub skip_private: bool,
+
+    /// Do not print the disassembled bytecodes of each function.
+    #[clap(long = "skip-code")]
+    pub skip_code: bool,
+
+    /// Do not print locals of each function.
+    #[clap(long = "skip-locals")]
+    pub skip_locals: bool,
+
+    /// Do not print the basic blocks of each function.
+    #[clap(long = "skip-basic-blocks")]
+    pub skip_basic_blocks: bool,
+
+    /// Treat input file as a script (default is to treat file as a module)
+    #[clap(short = 's', long = "script")]
+    pub is_script: bool,
+
+    /// The path to the bytecode file to disassemble; let's call it file.mv. We assume that two
+    /// other files reside under the same directory: a source map file.mvsm (possibly) and the Move
+    /// source code file.move.
+    #[clap(short = 'b', long = "bytecode")]
+    pub bytecode_file_path: String,
+
+    /// (Optional) Path to a coverage file for the VM in order to print trace information in the
+    /// disassembled output.
+    #[clap(short = 'c', long = "move-coverage-path")]
+    pub code_coverage_path: Option<String>,
+}
+
+#[async_trait]
+impl CliCommand<String> for AptosDisassembler {
+    fn command_name(&self) -> &'static str {
+        "AptosDisassembler"
+    }
+
+    async fn execute(self) -> CliTypedResult<String> {
+        let move_extension = MOVE_EXTENSION;
+        let mv_bytecode_extension = MOVE_COMPILED_EXTENSION;
+        let source_map_extension = SOURCE_MAP_EXTENSION;
+
+        let source_path = Path::new(&self.bytecode_file_path);
+        let extension = source_path
+            .extension()
+            .expect("Missing file extension for bytecode file");
```

Comment:
I see a lot of this in the code. Let's replace all of these with `.context()`, like this:
```suggestion
            .context("Missing file extension for bytecode file")?;
```
We should avoid panicking where possible and use Result instead. It's less of a big deal in CLIs but panics are still bad UX, the error messages are gross.

Reply:
Its actually quite a deal in a CLI as well -- never crash on user input, tools doing this are experienced as crappy.